theia-extensions/viewer-prototype/src/browser/trace-server-connection-status-client-impl.ts

```diff
@@ -2,13 +2,17 @@
 import { injectable } from 'inversify';
 import { TraceServerConnectionStatusClient } from '../common/trace-server-connection-status';
 
+type Listener = (serverStatus: boolean) => void;
 @injectable()
 export class TraceServerConnectionStatusClientImpl implements TraceServerConnectionStatusClient {
     protected active = false;
+    protected lastStatus = false;
+    protected listeners: Listener[] = [];
 
     updateStatus(status: boolean): void {
         if (this.active) {
-            TraceServerConnectionStatusClientImpl.renderStatus(status);
+            this.lastStatus = status;
+            this.listeners.forEach(fn => fn(status));
```

Comment:
please rename `fn` all the way down to `listener` or something. This would make the individual functions easier to read.

Reply:
Changed.